File: normal/pathSum.dart
// Definition for a binary tree node.
class TreeNode {
  int val;
  TreeNode? left;
  TreeNode? right;

  TreeNode(this.val, [this.left, this.right]);
}

class Solution {
  bool hasPathSum(TreeNode? root, int targetSum) {
    if (root == null) return false;

    // Check if it's a leaf node and the value matches the remaining targetSum.
    if (root.left == null && root.right == null && root.val == targetSum) {
      return true;
    }

    // Recurse for left and right subtrees with the updated targetSum.
    targetSum -= root.val;
    return hasPathSum(root.left, targetSum) || hasPathSum(root.right, targetSum);
  }
}

File: normal/pathSum.rs
// Definition for a binary tree node.
#[derive(PartialEq, Eq, Clone)]
pub struct TreeNode {
    pub val: i32,
    pub left: Option<Box<TreeNode>>,
    pub right: Option<Box<TreeNode>>,
}

impl Solution {
    pub fn has_path_sum(root: Option<Box<TreeNode>>, target_sum: i32) -> bool {
        if let Some(node) = root {
            // Check if it's a leaf node and the value matches the remaining targetSum.
            if node.left.is_none() && node.right.is_none() && node.val == target_sum {
                return true;
            }
            
            // Recurse for left and right subtrees with the updated targetSum.
            let new_target_sum = target_sum - node.val;
            return Solution::has_path_sum(node.left, new_target_sum) || 
                   Solution::has_path_sum(node.right, new_target_sum);
        }
        false
    }
}
